rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -733,6 +744,7 @@ mod tests {
         writer.set(prefix_2, key2, value2).await.unwrap();
 
         let flusher = writer.commit::<&str, Vec<u32>>().await.unwrap();
+        assert_eq!(Some(2_u64), flusher.total_keys());
```

Comment:
We need more testing around this.

Reply:
Agreed, wanted to make sure I wasn't missing the mark directionally before going deeper. Will write next